common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsManager.kt

```diff
@@ -69,14 +72,42 @@ class DiagnosticsManager(
 
     fun trackEvent(diagnosticsEvent: DiagnosticsEvent) {
         enqueue {
-            val anonymizedEvent = diagnosticsAnonymizer.anonymizeEventIfNeeded(diagnosticsEvent)
-            // WIP: Check that file size is not above certain limit. If it is, delete.
-            verboseLog("Tracking diagnostics event: $anonymizedEvent")
-            try {
-                diagnosticsFileHelper.appendEventToDiagnosticsFile(anonymizedEvent)
-            } catch (e: IOException) {
-                verboseLog("Error tracking diagnostics event: $e")
-            }
+            trackEventInCurrentThread(diagnosticsEvent)
+        }
+    }
+
+    private fun getEventsToSync(): List<JSONObject> {
+        val diagnosticsList = diagnosticsFileHelper.readDiagnosticsFile()
+        val diagnosticsInFileCount = diagnosticsList.size
+        if (diagnosticsInFileCount > MAX_NUMBER_EVENTS) {
+            val eventsToRemoveCount = diagnosticsInFileCount - MAX_NUMBER_EVENTS + 1
+            diagnosticsFileHelper.deleteOlderDiagnostics(eventsToRemoveCount)
+            trackMaxEventsStoredLimitReached(diagnosticsInFileCount, eventsToRemoveCount)
```

Comment:
Note that we currently perform this deleting + sending diagnostics event when syncing, not tracking. This means the file can potentially exceed the max number of events but it shouldn't be that much if we sync every time the SDK is configured. Also, note that if we fail syncing 3 times, we simply delete the whole file, so it shouldn't grow that much even if it keeps failing syncing. Lmk what you think!